<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>541</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QWidget" name="widget_2" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>200</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <widget class="QWidget" name="widget_4" native="true">
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QWidget" name="widget_6" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>查找目标</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="target">
               <property name="readOnly">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="widget_7" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>数据长度</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="length">
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="widget_9" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>最小值  </string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="min">
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="widget_10" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>最大值  </string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="max">
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="widget_8" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QPushButton" name="generate">
               <property name="text">
                <string>生成数据</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widget_5" native="true">
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QStackedWidget" name="stackedWidget">
            <property name="currentIndex">
             <number>1</number>
            </property>
            <widget class="QWidget" name="orderSearch">
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <widget class="QRadioButton" name="binary">
                <property name="text">
                 <string>二分查找</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="index">
                <property name="text">
                 <string>索引表查找</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="treeSearch">
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <widget class="QRadioButton" name="BTree">
                <property name="text">
                 <string>   B树</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="balancedTree">
                <property name="text">
                 <string>平衡二叉树</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="hashSearch">
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <widget class="QRadioButton" name="hash">
                <property name="text">
                 <string>拉链法</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QWidget" name="widget_3" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <widget class="QTabWidget" name="tabWidget">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>771</width>
         <height>521</height>
        </rect>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="currentIndex">
        <number>1</number>
       </property>
       <widget class="QWidget" name="order">
        <property name="contextMenuPolicy">
         <enum>Qt::CustomContextMenu</enum>
        </property>
        <property name="accessibleName">
         <string/>
        </property>
        <attribute name="title">
         <string>顺序表</string>
        </attribute>
        <widget class="BtnGroup" name="btnGroup" native="true">
         <property name="geometry">
          <rect>
           <x>9</x>
           <y>9</y>
           <width>466</width>
           <height>34</height>
          </rect>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
         </layout>
        </widget>
        <widget class="ListSearch" name="showBoard" native="true">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>50</y>
           <width>741</width>
           <height>431</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">#LoginWin
{

border:1px solid #014F84;
}</string>
         </property>
        </widget>
       </widget>
       <widget class="QWidget" name="tree">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <attribute name="title">
         <string>树表</string>
        </attribute>
        <widget class="BtnGroup" name="btnGroup_2" native="true">
         <property name="geometry">
          <rect>
           <x>9</x>
           <y>9</y>
           <width>466</width>
           <height>34</height>
          </rect>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
         </layout>
        </widget>
        <widget class="TreeSearch" name="showBoard_2" native="true">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>50</y>
           <width>741</width>
           <height>431</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">#LoginWin
{

border:1px solid #014F84;
}</string>
         </property>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_2">
        <attribute name="title">
         <string>散列表</string>
        </attribute>
        <widget class="BtnGroup" name="btnGroup_3" native="true">
         <property name="geometry">
          <rect>
           <x>9</x>
           <y>9</y>
           <width>466</width>
           <height>34</height>
          </rect>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
         </layout>
        </widget>
        <widget class="HashSearch" name="showBoard_3" native="true">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>50</y>
           <width>741</width>
           <height>431</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">#LoginWin
{

border:1px solid #014F84;
}</string>
         </property>
        </widget>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ListSearch</class>
   <extends>QWidget</extends>
   <header location="global">listsearch.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>HashSearch</class>
   <extends>QWidget</extends>
   <header>hashsearch.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TreeSearch</class>
   <extends>QWidget</extends>
   <header>treesearch.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>BtnGroup</class>
   <extends>QWidget</extends>
   <header>btngroup.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
